### PR TITLE
Fix a race condition bug that does not store Trial Job cancel status correctly

### PR DESCRIPTION
**Pull request by yds05**
1. For PAITrainingService, we should rely on the status returned from PAI rest server to change trial job's status, rather than setting it's cancelling status in cancelTrialJob() method, because jobInfoCollector will then reset job's status based on PAI rest server's result. 
2. For RemoteMachineTrainingService, it almost the same, we should not set trial's status in cancelTrialJob() method.

### CODE REVIEW

File: src/nni_manager/training_service/pai/paiTrainingService.ts [hunk 1]
```diff
@@ -324,18 +324,19 @@ class PAITrainingService implements TrainingService {
...
             } else {
-                if (isEarlyStopped) {
-                    trialJobDetail.status = 'EARLY_STOPPED';
-                }
                 deferred.resolve();
             }
         });
 
+        // Set trialjobDetail's early stopped field, to mark the job's cancellation source
+        trialJobDetail.isEarlyStopped = isEarlyStopped;
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/pai/paiTrainingService.ts]: I think this line has to be put at the beginning of this function.

File: src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts [hunk 2]
```diff
@@ -280,7 +280,9 @@ class RemoteMachineTrainingService implements TrainingService {
             const jobpidPath: string = this.getJobPidPath(trialJob.id);
             try {
                 await SSHClientUtility.remoteExeCommand(`pkill -P \`cat ${jobpidPath}\``, sshClient);
-                trialJob.status = getJobCancelStatus(isEarlyStopped);
+                //TODO: delete and move set USER_CANCELLED/EARLY_STOP in getTrialJob
+                // Mark the toEarlyStop tag here
+                trialJob.isEarlyStopped = isEarlyStopped;
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts]: this line should be put before the `await SSHClientUtility` (i.e., line 282)

File: src/nni_manager/training_service/pai/paiTrainingService.ts (hunk 1 above)
Comment (yds05) [reply src/nni_manager/training_service/pai/paiTrainingService.ts]: Not very sure why we should put this line to the beginning. 
The beginning of this function has many condition checking. If we put this line to the beginning of this function but condition checking failed, the trial job status may be wrong.

File: src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts (hunk 2 above)
Comment (yds05) [reply src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts]: Done

File: src/nni_manager/training_service/pai/paiTrainingService.ts (hunk 1 above)
Comment (QuanluZhang) [reply src/nni_manager/training_service/pai/paiTrainingService.ts]: agree, then we should put this line before line 328 `request(...`

Comment (yds05) [reply src/nni_manager/training_service/pai/paiTrainingService.ts]: Yes Sure.

**Pull request merged**